site/docs/latest/admin/http.md

```diff
@@ -0,0 +1,317 @@
+---
+title: BookKeeper http endpoints
+---
+
+This document is to introducing BookKeeper http endpoints that could be used for BookKeeper administration.
+If user want to use this feature, parameter "httpServerEnabled" need to be set to "true" in file conf/bk_server.conf.
+
+## All the endpoints
+
+Currently all the http endpoints could be divided into these 4 components:
+1. Heartbeat: heartbeat for a specific bookie.
+1. Config: doing the server configuration for a specific bookie.
+1. Ledger: http endpoints related to ledgers.
+1. Bookie: http endpoints related to bookies.
+1. AutoRecovery: http endpoints related to auto recovery.
+
+## Heartbeat
+
+### Endpoint: /heartbeat
+* Method: GET
+* Description: Get heartbeat status for a specific bookie
+* Response: 
+
+| Code   | Description |
+|:-------|:------------|
+|200 | Successful operation |
+
+## Config
+
+### Endpoint: /api/v1/config/server_config
+1. Method: GET
+    * Description: Get value of all configured values overridden on local server config
+    * Response:  
+    
+        | Code   | Description |
+        |:-------|:------------|
+        |200 | Successful operation |
+        |403 | Don't have permission |
+        |404 | Error handling |
+1. Method: PUT
```

Comment:
Is there already an admin command to do this? Does this access a config in zookeeper or on the local disk?

Reply:
seems not command for this. it only update the inmemory value, which could be used for later ops.

Reply:
I'm not sure it's wise to expose functionality like this at all. It's just asking for people to change configuration in an uncontrolled and unpredictable manner. Was there a driving use case to add this to the http endpoints?

Reply:
@ivankelly I think http endpoints (and configuration service) was introduced by twitter initially and Jia extended it to cover all admin commands. I can see this is useful on dynamic configurations.  but Twitter folks can chime in more on the use cases.

BTW If there is a question on a feature whether this should be included or not, we should probably be discussing a bit earlier on the coding pull request, rather than later on the documentation pull request. (for future improvements)